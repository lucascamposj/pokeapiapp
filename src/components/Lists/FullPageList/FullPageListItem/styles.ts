import styled from 'styled-components/native';

export const Container = styled.TouchableOpacity`
  padding: 15px 0;
  border-bottom-width: 1px;
  border-bottom-color: ${({ theme }) => theme.colors.common.gray};
`;

export const ItemText = styled.Text`
  text-transform: capitalize;
  font-family: 'RobotoMono-Medium';
  font-size: 19px;
`;
