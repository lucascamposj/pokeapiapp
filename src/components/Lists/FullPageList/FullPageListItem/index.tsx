import React from 'react';
import { GestureResponderEvent, TouchableOpacityProps } from 'react-native';

import { Container, ItemText } from './styles';

export interface ListItemProps extends TouchableOpacityProps {
  name: string;
  url: string;
  onPressItem: (
    event: GestureResponderEvent,
    name: string,
    url: string,
  ) => void;
}

const FullPageListItem: React.FC<ListItemProps> = ({
  name,
  url,
  onPressItem,
  ...rest
}: ListItemProps) => {
  return (
    <Container onPress={(e) => onPressItem(e, name, url)} {...rest}>
      <ItemText>{name}</ItemText>
    </Container>
  );
};

export default FullPageListItem;
