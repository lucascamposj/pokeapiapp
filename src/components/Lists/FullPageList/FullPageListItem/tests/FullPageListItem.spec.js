import React from 'react';
import {shallow} from 'enzyme';
import FullPageListItem from '../../FullPageListItem';

describe('FullPageListItem', () => {
    describe('Rendering', () => {
        it('should match to snapshot loaded', () => {
            const mockPressItem = jest.fn();
            const component = shallow(
              <FullPageListItem
                name={'Test nome'}
                onPressItem={mockPressItem}
                url="url-test.com"
              />)
            expect(component).toMatchSnapshot()
        });
    });
});
