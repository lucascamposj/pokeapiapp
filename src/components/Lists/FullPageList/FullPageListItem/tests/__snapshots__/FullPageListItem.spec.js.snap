// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullPageListItem Rendering should match to snapshot loaded 1`] = `
<Styled(Component)
  onPress={[Function]}
>
  <Styled(Text)>
    Test nome
  </Styled(Text)>
</Styled(Component)>
`;
